apis_core/generic/views.py

```diff
@@ -0,0 +1,202 @@
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.contrib.auth import get_permission_codename
+from django.views.generic import DetailView
+from django.views.generic.base import TemplateView
+from django.views.generic.edit import CreateView, UpdateView, DeleteView
+from django.urls import reverse
+from django.forms import modelform_factory
+
+from django_filters.views import FilterView
+from django_tables2 import SingleTableMixin
+from django_tables2.tables import table_factory
+from django_tables2.columns import library
+from dal import autocomplete
+
+from .tables import GenericTable
+from .filtersets import filterset_factory, GenericFilterSet
+from .forms import GenericModelForm
+from .helpers import first_match_via_mro, template_names_via_mro, generate_search_filter
+
+
+class Overview(TemplateView):
+    template_name = "generic/overview.html"
+
+
+class GenericModelMixin:
+    """
+    A mixin providing the common functionality for allthe views working
+    with `generic` models - that is models that are accessed via the
+    contenttype framework (using `app_label.model`)
+    It sets the `.model` of the view and generates a list of possible template
+    names (based on the MRO of the model).
+    If the view has a `permission_action_required` attribute, this is used
+    to set the permission required to access the view for this specific model.
+    """
+
+    def setup(self, *args, **kwargs):
+        super().setup(*args, **kwargs)
+        self.model = kwargs.get("contenttype").model_class()
+        self.queryset = self.model.objects.all()
+
+    def get_template_names(self):
+        template_names = super().get_template_names()
+        suffix = self.template_name_suffix + ".html"
+        additional_templates = template_names_via_mro(self.model, suffix) + [
+            f"generic/generic{suffix}"
+        ]
+        template_names += filter(
+            lambda template: template not in template_names, additional_templates
+        )
+        return template_names
+
+    def get_permission_required(self):
+        if hasattr(self, "permission_action_required"):
+            return [
+                get_permission_codename(
+                    self.permission_action_required, self.model._meta
+                )
+            ]
+        return []
+
+
+class List(GenericModelMixin, PermissionRequiredMixin, SingleTableMixin, FilterView):
+    """
+    List view for a generic model
```

Comment:
I would recommend ending full sentences with the appropriate punctuation, i.e. full stops.

This not only helps readability overall but also prevents IDEs from flagging blocks of texts for failing language checks.

My IDE interprets _all_ the Docstrings in these files to contain run-on sentences (!) and it flagged this particular one for nonsensical grammar because it thinks the uncountable noun "access" was given an indefinite article ("a generic model access").